Fix incorrect buffer index failure

ilya-kupchenko opened the pull request:
Description:
bug=fixes #3075 #3076 #3077

Problem
MicroAllocator does not validate the buffer indices. If a model provides an out-of-bounds buffer index for an input, output, or intermediate tensor, it results in segmentation fault.

Solution
Add validation inside MicroAllocator to check that the buffer index for every tensor. If an invalid index is found, InitializeTfLiteTensorFromFlatbuffer and InitializeTfLiteEvalTensorFromFlatbuffer will now return kTfLiteError.

Files Modified
    tensorflow/lite/micro/micro_interpreter.cc
    tensorflow/lite/micro/micro_interpreter_test.cc

Testing
Added new Bazel tests that use models with invalid buffer-to-tensor mappings.

gonnet commented on tensorflow/lite/micro/micro_allocator.cc:
```diff
@@ -230,6 +230,19 @@ TfLiteStatus InitializeTfLiteTensorFromFlatbuffer(
+  const uint32_t buffer_index = flatbuffer_tensor.buffer();
+  if (buffer_index >= buffers->size()) {
+    MicroPrintf(
+        "Tensor references invalid buffer index %u, model has only %d buffers",
```
All the other calls to `MicroPrintf()` have strings ending in `\n`.

ilya-kupchenko commented:
I squashed the commit messages, fixed the style and added end-of-line symbols, just in case. Some MicroPrintf() calls in micro_allocator.cc have them, some not. Let me know if you want me to add/remove them for all MicroPrintf() calls it in all the project in a unified way. 
If you allow, I will fix #3071, #3072, #3073, #3074, #3128and #2721. Let me know if this is OK (or what other things I can be useful for).
PS. While trying to run the CI/CD tests I ran into some errors. I described them here #3240. If this is a bug, I can push the patch for it.
Thank you!
